File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/wrapping_and_braces/JavaForStatementSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.wrapping_and_braces;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavaForStatementSettings.class)
public interface JavaForStatementSettings {

  Optional<Wrapping> wrapping();

  Optional<Force> forceBraces();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/wrapping_and_braces/JavaForStatementSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.wrapping_and_braces

import com.fasterxml.jackson.annotation.JsonProperty

data class JavaForStatementSettings(

    @JsonProperty("Wrapping")
    val wrapping: Wrapping?,

    @JsonProperty("Force braces")
    val forceBraces: Force?
)
